packages/smithy-core/src/smithy_core/exceptions.py

```diff
@@ -1,39 +1,96 @@
 #  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 #  SPDX-License-Identifier: Apache-2.0
-class SmithyException(Exception):
+from dataclasses import dataclass, field
+from typing import Literal
+
+
+class SmithyError(Exception):
     """Base exception type for all exceptions raised by smithy-python."""
 
 
-class SerializationException(Exception):
+type Fault = Literal["client", "server"] | None
+"""Whether the client or server is at fault.
+
+If None, then there was not enough information to determine fault.
+"""
+
+
+@dataclass(kw_only=True)
+class CallError(SmithyError):
+    """Base exception to be used in application-level errors.
+
+    Implements :py:class:`.interfaces.retries.ErrorRetryInfo`.
+    """
+
+    fault: Fault = None
+    """Whether the client or server is at fault.
+
+    If None, then there was not enough information to determine fault.
+    """
+
+    message: str = field(default="", kw_only=False)
+    """The message of the error."""
+
+    is_retry_safe: bool | None = None
+    """Whether the exception is safe to retry.
+
+    A value of True does not mean a retry will occur, but rather that a retry is allowed
+    to occur.
+
+    A value of None indicates that there is not enough information available to
+    determine if a retry is safe.
+    """
+
+    retry_after: float | None = None
+    """The amount of time that should pass before a retry.
+
+    Retry strategies MAY choose to wait longer.
+    """
+
+    is_throttling_error: bool = False
+    """Whether the error is a throttling error."""
+
+    def __post_init__(self):
+        super().__init__(self.message)
+
+
+@dataclass(kw_only=True)
+class ModeledError(CallError):
+    """Base exception to be used for modeled errors."""
+
+    fault: Fault = "client"
```

Comment:
Why default to `"client"` instead of None?

Reply:
Modeled errors will always have this defined as the Smithy trait requires it. So this isn't really important either way